.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,36 @@
+name: Differential ShellCheck
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
Likely only want this on the dev branch

Reply:
I think the ideal configuration for this repo is as follows:

```yml
on:
  push:
    branches: [ master, dev, bugfix ]
  pull_request:
    branches: [ dev, bugfix ]
```

When running on push, all defects are reported in your Security Dashboard on GitHub so that you will see all defects on those branches.
From what I see, you use branches `dev` and `bugfix` for development and fixes, so I think scanning incoming PRs to these branches might be beneficial.

@Maffooch Please let me know if this works for you or if you want only `dev` to be scanned. Thank you.

Reply:
Good catch. I think the following would be ideal then. Pushes to master mainly happen when merging dev/bugfix during a release
```
on:
  push:
    branches: [ dev, bugfix ]
  pull_request:
    branches: [ dev, bugfix ]
```

Reply:
I have applied your suggestion.